mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -105,25 +110,28 @@ private boolean isMessageWithOnlyOneAttachment(Optional<ContentDispositionField>
         return contentDisposition.isPresent() && contentDisposition.get().isAttachment();
     }
 
-    private List<MessageAttachment> listAttachments(Multipart multipart, Context context) throws IOException {
-        ImmutableList.Builder<MessageAttachment> attachments = ImmutableList.builder();
+    private Stream<MessageAttachment> listAttachments(Multipart multipart, Context context) throws IOException {
+        ThrowingFunction<Entity, Stream<MessageAttachment>> listAttachments = Throwing.function(entity -> listAttachments(entity, context));
```

Comment:
you don't need to do that, you can just remove IOException from listAttachments signature

Reply:
Well catched!!!